bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/AutoRecoveryMain.java

```diff
@@ -85,40 +85,36 @@ public AutoRecoveryMain(ServerConfiguration conf, StatsLogger statsLogger)
             throws IOException, InterruptedException, KeeperException, UnavailableException,
             CompatibilityException {
         this.conf = conf;
-        Set<Watcher> watchers = new HashSet<Watcher>();
-        // TODO: better session handling for auto recovery daemon  https://issues.apache.org/jira/browse/BOOKKEEPER-594
```

Comment:
yeah, it doesn't really make sense to shutdown auto recovery when expired from zookeeper, especially if the underlying zookeeper is reconnectable. I am not worried about replication worker and the normal zookeeper metadata operations in auditor, however can you explain a bit after removing this part, how does Auditor elector handles the expire and retries?

Reply:
There is a logic in AuditorElector will shutdown itself when session expired : https://github.com/apache/bookkeeper/blob/master/bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/AuditorElector.java#L203

when AuditorElector shuts itself down, the death watcher will quit and then AutoRecoveryMain will quit. so the session expire logic in AutoRecoveryMain is not really needed.